### ISSUE TITLE
Is there a way to wait for window closure?

### ISSUE THREAD

**Issue by viraptor** [FlaUI/FlaUI#484]
**Describe the bug**
I may be missing something obvious, but I can't find a way to wait for window to be close.
I'm using UI3.

**Code snippets**

I've tried the following:

This never exits - `window` seems to be available even though it gets closed.
```
Retry.WhileTrue(() => window.IsAvailable, TimeSpan.FromSeconds(10), TimeSpan.FromMilliseconds(100), true);
```

This fails with exception `FlaUI.Core.Exceptions.PropertyNotSupportedException: The requested property 'IsOffscreen [#30022]' is not supported`
```
Retry.WhileFalse(() => window.IsOffscreen, TimeSpan.FromSeconds(10), TimeSpan.FromMilliseconds(100), true);
```

I can wait until I can't find the window in the parent window anymore, but that's not great - I'd like to write controller for a specific window with a function "Confirm" which clicks OK, then waits for the window itself to disappear.

**Comment by LevYas** [comment]
I have a case when I need to close my app and then open it and check if everything is OK.
For that, I have a Stop method, which does this:

       ((Window?)MainWindow)?.Close();
        _app?.Close();
        _app?.Dispose();
        _automation.Dispose();

After it completes, I start the app immediately, and it works without any problems.
Hope it helps somehow.

**Comment by dfev77** [comment]
Yes Window parent can be null. There could be other cases there the properties could throw exceptions....

To wait for window close we do a retry until the method below returns false (there are also a couple of exceptions to take care of as well)
`
internal static bool Exists(this Window window)
        {
            Contract.NotNull(window, nameof(window));
            IntPtr windowHandle;

            try
            {
                windowHandle = window.Properties.NativeWindowHandle.ValueOrDefault;
            }
            catch (Exception ex) when (ex is COMException ||
                                       ex is ArgumentException)
            {
                // control no longer exists
                windowHandle = IntPtr.Zero;
            }

            return IsWindow(windowHandle);
        }
`
`IsWindow` comes from `PInvoke.User32`

**Comment by Roemer** [comment]
Your 3rd approach would be the correct one: Search for the window in the retry and as long as it can be found, it is still there. Can't speak about the method from @dfev77 but if that works, that could be added to FlaUI (probably to the base automation element).

**Comment by n9** [comment]
@dfev77 `IsWindow` might not be 100% reliable:

> A thread should not use IsWindow for a window that it did not create ... because window handles are recycled the handle could even point to a different window.

(source https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-iswindow#remarks)
